src/service/group-service.js

```diff
@@ -0,0 +1,86 @@
+import database from '../config/database';
+import ServiceError from '../errors/service-error';
+
+export default class GroupService {
+  constructor(groupModel, groupDataMapper, userModel) {
+    this.groupModel = groupModel;
+    this.groupDataMapper = groupDataMapper;
+    this.userModel = userModel;
+  }
+
+  async add(groupDTO) {
+    const group = this.groupDataMapper.toDalEntity(groupDTO);
+
+    try {
+      return this.groupModel.create(group);
+    } catch (e) {
+      throw new ServiceError(e.message);
+    }
+  }
+
+  async addUsersToGroup(groupId, userIds) {
+    const transaction = await database.transaction();
+
+    try {
+      const group = await this.groupModel.findByPk(groupId);
+      const users = await Promise.all(
+        userIds.map(async (userId) => this.userModel.findByPk(userId)),
+      );
+
+      if (!group || !users) {
```

Comment:
as users is an array (map always returns an array) you should also check if it is not empty
`!users || !users.length)`